libdebug/architectures/amd64/amd64_syscall_hijacker.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is part of libdebug Python library (https://github.com/libdebug/libdebug).
+# Copyright (c) 2024 Gabriele Digregorio. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for details.
+#
+
+from typing import Any, Callable
+from libdebug.state.thread_context import ThreadContext
+from libdebug.architectures.syscall_hijacking_manager import SyscallHijackingManager
+
+
+class Amd64SyscallHijacker(SyscallHijackingManager):
```

Comment:
Right now I believe that this file is architecture-independent, thus we should rename it GenericSyscallHijacker or something like that.

Reply:
let us continue in the argx discussion, since that is the point